test/unit/ObjectMakrTests/getFullProperties.js

```diff
@@ -0,0 +1,58 @@
+define(["mocks"], function (mocks) {
+    return function () {
+        var expect = require("chai").expect;
+
+        it("returns the full properties", function () {
+            // Arrange
+            var objectMaker = mocks.mockObjectMakr();
+
+            var fullProperties = {
+                    "Animal": {
+                        "name": "",
+                        "weight": 0,
+                        "age": 0
+                    },
+                    "Bird": {
+                        "name": "",
+                        "weight": 0,
+                        "age": 0
+                    },
+                    "Penguin": {
+                        "name": "",
+                        "weight": 0,
+                        "age": 0,
+                        "onMake": function(penguin){
```

Comment:
Nit (here and elsewhere): should be a space after the `)` in function definitions